piff/gp_interp.py

```diff
@@ -13,113 +13,127 @@
-                        Normally, the parameters being interpolated are not mean 0, so you would
-                        want this to be True, but if your parameters have an a priori mean of 0,
-                        then subtracting off the realized mean would be invalid.  [default: True]
-    :param logger:      A logger object for logging debug info. [default: None]
+    An interpolator that uses gaussian process from treegp to interpolate multiple surfaces.
+    :param keys:         A list of star attributes to interpolate from. Must be 2 attributes
+                         using two-point correlation function to estimate hyperparameter(s).
+    :param kernel:       A string that can be evaled to make a
+                         sklearn.gaussian_process.kernels.Kernel object.  The reprs of
+                         sklearn.gaussian_process.kernels will work, as well as the repr of a
+                         custom treegp VonKarman object.  [default: 'RBF(1)']
+    :param optimize:     Boolean indicating whether or not to try and optimize the kernel by
+                         computing the two-point correlation function or a maximum likelihood.
+                         [default: True]
+    :param optimizer:    Indicates which technics to used for optimizing the kernel. Two options
+                         are available. "two-pcf" optimize the kernel on the 1d/2d 2-point correlation
+                         function estimate by treecorr. "log-likelihood" used the classical
+                         gaussian process maximum likelihood to optimize the kernel. [default: "two-pcf"]
+    :param anisotropic:  2D 2-point correlation function. Used 2D correlation function for the
+                         fiting part of the GP instead of a 1D correlation function. Used only
+                         if optimizer is "two-pcf", for "log-likelihood" will do an anisotropic fit
+                         if the kernel is anisotropic. [default: False]
+    :param p0:           Initial guess for correlation length, and quantity of anisotropy
+                         (e1 and e2 params as for galaxy shape). Used only if optimizer is "two-pcf"
+                         and anisotropic is True. [default: [3000., 0., 0.]]
+    :param normalize:    Whether to normalize the interpolation parameters to have a mean of 0.
+                         Normally, the parameters being interpolated are not mean 0, so you would
+                         want this to be True, but if your parameters have an a priori mean of 0,
+                         then subtracting off the realized mean would be invalid.  [default: True]
+    :param white_noise:  A float value that indicate the ammount of white noise that you want to
+                         use during the gp interpolation. This is an additional uncorrelated noise
+                         added to the error of the PSF parameters. [default: 0.]
```

Comment:
I am not sure to understand the comment; you want me to say that the white noise is an uncorrelated additional noise? For the variance I am not so sure, it could be great for the purpose that you proposed, but I am afraid that it could be used in the wrong way.